### PR TITLE
fedekunze/1977-fix ledger errors

### LINKED ISSUE #1977: Ledger error notification body is always {}
screen shot 2019-02-13 at 10 23 40 am

### PR DESCRIPTION
**Pull request by fedekunze**
Closes #1977 

**Description:**

Fixes error bug and **makes ledger available on production**

screen shot 2019-02-15 at 1 34 11 pm

Thank you! 🚀

---

For contributor:

- [x] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [x] Reviewed `Files changed` in the github PR explorer
- [x] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: app/src/renderer/components/common/TmSessionHardware.vue [hunk 1]
```diff
@@ -2,11 +2,11 @@
   <div class="tm-session">
     <div class="tm-session-container">
       <div class="tm-session-header">
-        <a @click="setState('welcome')"
-          ><i class="material-icons">arrow_back</i></a
-        >
+        <a @click="setState('welcome')">
+          <i class="material-icons">arrow_back</i>
```
Comment (sabau) [inline app/src/renderer/components/common/TmSessionHardware.vue]: <3

File: app/src/renderer/components/common/TmSessionHardware.vue [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
+        </a>
         <div class="tm-session-title">Sign In</div>
-        <a @click="help"><i class="material-icons">help_outline</i></a>
+        <a @click="help"> <i class="material-icons">help_outline</i> </a>
```
Comment (sabau) [inline app/src/renderer/components/common/TmSessionHardware.vue]: given we will soon face cross browser stuff, I'll suggest to enforce spacing with css if that's the need here

File: test/unit/specs/store/ledger.spec.js [hunk 3]
```diff
@@ -184,14 +190,16 @@ describe(`Module: Ledger`, () => {
 
         it(`sets an error on failure`, async () => {
           const pubKey = Buffer.from([1])
-          state.cosmosApp.publicKey = jest.fn(() =>
-            Promise.reject(`Bad Key handle`)
+          state.cosmosApp.publicKey = jest.fn(async () =>
```
Comment (sabau) [inline test/unit/specs/store/ledger.spec.js]: This form shuld be equivalent but seems more readable in this case:
```
          state.cosmosApp.publicKey = () =>
            Promise.resolve({
              error_message: `Bad key handle`
            })
```

File: test/unit/specs/store/ledger.spec.js [hunk 4]
```diff
@@ -214,7 +222,9 @@ describe(`Module: Ledger`, () => {
         it(`fails if message is not JSON`, async () => {
           const msg = `{"account_number": 1,"chain_id": "some_chain","fee": {"amount": [{"amount": 10, "denom": "DEN"}],"gas": 5},"memo": "MEMO","msgs": ["SOMETHING"],"sequence": 3}`
           state.cosmosApp.sign = jest.fn(async () =>
-            Promise.reject(`Bad key handle`)
+            Promise.resolve({
```
Comment (sabau) [inline test/unit/specs/store/ledger.spec.js]: Similar to the previous:
```
          state.cosmosApp.sign = () =>
            Promise.resolve({
              error_message: `Bad key handle`
            })
```

File: test/unit/specs/store/ledger.spec.js [hunk 5]
```diff
@@ -123,13 +129,13 @@ describe(`Module: Ledger`, () => {
         })
 
         it(`fails if one of the function throws`, async () => {
-          dispatch = jest.fn(async () => Promise.reject(`error`))
+          dispatch = jest.fn(async () => Promise.reject(new Error(`error`)))
```
Comment (sabau) [inline test/unit/specs/store/ledger.spec.js]: Probably here there are some other occurencies of `jest.fn(async` ... `)` that can be dropped until you are not expecting something from `dispatch` using a simple `()` will work

**Comment by faboweb** [comment]
@sabau Did you manually test the change?

**Pull request merged**